openedx/frontend-app-course-authoring
# Move frontend-lib-content-components code into this repo [FC-0062]

## Description

The `frontend-lib-content-components` library is only used by this MFE, and having it in a separate repository slows down development (and installation, maintenance / version bumps, etc.). This PR moves the code into this repo.

## Supporting information

Discussed on Slack: https://openedx.slack.com/archives/C04BM6YC7A6/p1722545083239339

Relates to https://github.com/openedx/frontend-lib-content-components/issues/499

## What has actually changed?

Most of the code has not changed! The changes I've made are limited to the diff you can see in the most recent few commits:
* d3d5fe0e1bb438717d10b1c09f58cd8748add67c Removed unneeded files from lib-content-components
* 28e1956708907f479aa994fb10064ab4a20151e2 chore: update imports, fix lint issues
* etc. - see complete list below

## What are the benefits?

In addition to **making development much simpler**, this has a **major impact on reducing bundle size**:

| Before: 10 MB | After: 7 MB |
|--|--|
| ![before](https://github.com/user-attachments/assets/57ab678c-db9b-4594-9a73-f3c02cbd472e) |  ![after](https://github.com/user-attachments/assets/98105622-bf99-46d8-866a-e8c1ff2084b0) |


## Testing instructions

Check out this branch, run `npm install`, and Ensure that editing Text, Problem, and Video components continues to work as normal.

## Other information

I have copied the entire git history of `frontend-lib-content-components`, so that we don't lose it. That will create a second root/initial commit in this repo, but I believe that's OK.

Before we merge, it may be necessary to move some localizations in transifex. I'm not sure how to do that.
